File: python_solutions/58.py
class Solution:
    def lengthOfLastWord(self, s: str) -> int:
        idx = 0
        prev_word_len = 0
        while idx < len(s):
            while idx < len(s) and s[idx] == ' ':
                idx += 1
            if idx < len(s) and s[idx] != ' ':
                prev_word_len = 0
                while idx < len(s) and s[idx] != ' ':
                    prev_word_len += 1
                    idx += 1
        return prev_word_len
        

File: c++_solutions/58.cpp
class Solution {
public:
    int lengthOfLastWord(string s) {
        size_t prev_word_len = 0;
        for (auto it = s.begin() ; it != s.end() ;){
            while (it != s.end() && *it == ' '){
                it++;
            }
            if (it != s.end()){
                prev_word_len = 0;
                while(it != s.end() && *it != ' '){
                    prev_word_len++;
                    it++;
                }
            }
        }
        return prev_word_len ;
        
    }
};
